config/routes.rb

```diff
@@ -13,20 +13,34 @@
   end
 
   namespace :editor do
-    root 'welcome#index'
+    get "/(:controller_name)/(:id)/(:action)", to: redirect {|params, request|
+      base_url = "#{request.scheme}://#{request.host_with_port}/sites/#{request.host}/editor"
+      path = [params[:controller_name], params[:id], params[:action]].compact.join("/")
+      if path.empty?
+        base_url
+      else
+        "#{base_url}/#{path}"
+      end
+    }
+  end
```

Comment:
Can you make this redirecting definition more descriptive? It will take a time to understand what it does. I will also try to think about that.